File: network/pi_prime_network.php
<?php

class PiPrimeNetwork {
    private $socket;

    public function __construct($port) {
        $this->socket = socket_create(AF_INET, SOCK_STREAM, SOL_TCP);
        socket_bind($this->socket, "127.0.0.1", $port);
        socket_listen($this->socket);
    }

    public function listen() {
        while (true) {
            $client = socket_accept($this->socket);
            $this->handleConnection($client);
        }
    }

    private function handleConnection($client) {
        $input = socket_read($client, 16);
        $input = unpack("N*", $input);
        $output = $this->isPrime($input[1]);
        socket_write($client, pack("C", $output));
    }

    private function isPrime($n) {
        if ($n < 2) return false;
        for ($i = 2; $i * $i <= $n; $i++) {
            if ($n % $i === 0) return false;
        }
        return true;
    }
}

$network = new PiPrimeNetwork(8080);
$network->listen();

?>

File: network/pi_prime_network.go
package main

import (
	"fmt"
	"math/big"
	"net"
)

type PiPrimeNetwork struct {
	listener net.Listener
}

func NewPiPrimeNetwork(port int) (*PiPrimeNetwork, error) {
	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
	if err != nil {
		return nil, err
	}
	return &PiPrimeNetwork{listener: listener}, nil
}

func (n *PiPrimeNetwork) Listen() error {
	for {
		conn, err := n.listener.Accept()
		if err != nil {
			return err
		}
		go n.handleConnection(conn)
	}
}

func (n *PiPrimeNetwork) handleConnection(conn net.Conn) {
	defer conn.Close()
	var input big.Int
	err := binary.Read(conn, binary.LittleEndian, &input)
	if err != nil {
		fmt.Println(err)
		return
	}
	output, err := isPrime(input)
	if err != nil {
		fmt.Println(err)
		return
	}
	err = binary.Write(conn, binary.LittleEndian, output)
	if err != nil {
		fmt.Println(err)
		return
	}
}

func isPrime(n big.Int) (bool, error) {
	if n.BitLen() < 2 {
		return false, nil
	}
	for i := big.NewInt(2); i.Cmp(n) <= 0; i.Add(i, big.NewInt(1)) {
		if n.Mod(n, i).Cmp(big.NewInt(0)) == 0 {
			return false, nil
		}
	}
	return true, nil
}

func main() {
	network, err := NewPiPrimeNetwork(8080)
	if err != nil {
		fmt.Println(err)
		return
	}
	err = network.Listen()
	if err != nil {
		fmt.Println(err)
		return
	}
}
